addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2325,6 +2327,23 @@ var SnippetsMenu = Widget.extend({
     _onUserValueWidgetClosing: function () {
         this.el.classList.remove('o_we_backdrop');
     },
+    /**
+     * Adds a loading overlay over the editor when option activated.
+     * 
+     * @private
+     * @param {OdooEvent} ev
+     */
+    _onStartLoadingOverlay: function (ev) {
+        this.$el.block(ev.data);
+    },
```

Comment:
There should not be any need to block the UI itself, just adding an effect *over the content* to indicate the option is being applied.